src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -430,22 +433,42 @@ public DynamicJsonValue ToJson()
             }
         }
 
-        public static IEnumerable<ResendItem> GetResendItems(ClusterOperationContext context, string database, long id)
+        public static IEnumerable<ResendItem> GetResendItemsForSubscriptionId(ClusterOperationContext context, string database, long id)
         {
             var subscriptionState = context.Transaction.InnerTransaction.OpenTable(ClusterStateMachine.SubscriptionStateSchema, ClusterStateMachine.SubscriptionState);
             using (SubscriptionConnectionsState.GetDatabaseAndSubscriptionPrefix(context, database, id, out var prefix))
             using (Slice.External(context.Allocator, prefix, out var prefixSlice))
             {
-                var resendItem = new ResendItem();
-
                 foreach (var item in subscriptionState.SeekByPrimaryKeyPrefix(prefixSlice, Slices.Empty, 0))
                 {
-                    resendItem.Type = (SubscriptionType)item.Key[prefixSlice.Size];
-                    resendItem.Id = item.Value.Reader.ReadStringWithPrefix((int)ClusterStateMachine.SubscriptionStateTable.Key, prefix.Length + 2);
-                    resendItem.ChangeVector = item.Value.Reader.ReadString((int)ClusterStateMachine.SubscriptionStateTable.ChangeVector);
-                    resendItem.Batch = Bits.SwapBytes(item.Value.Reader.ReadLong((int)ClusterStateMachine.SubscriptionStateTable.BatchId));
+                    yield return new ResendItem
+                    {
+                        Type = (SubscriptionType)item.Key[prefixSlice.Size],
+                        Id = item.Value.Reader.ReadStringWithPrefix((int)ClusterStateMachine.SubscriptionStateTable.Key, prefix.Length + 2),
+                        ChangeVector = item.Value.Reader.ReadString((int)ClusterStateMachine.SubscriptionStateTable.ChangeVector),
+                        Batch = Bits.SwapBytes(item.Value.Reader.ReadLong((int)ClusterStateMachine.SubscriptionStateTable.BatchId)),
+                        SubscriptionId = id
+                    };
+                }
+            }
+        }
 
-                    yield return resendItem;
+        public static IEnumerable<ResendItem> GetResendItemsForDatabase(ClusterOperationContext context, string database)
+        {
+            var subscriptionState = context.Transaction.InnerTransaction.OpenTable(ClusterStateMachine.SubscriptionStateSchema, ClusterStateMachine.SubscriptionState);
+            using var _ = Slice.From(context.Allocator, database.ToLowerInvariant(), SpecialChars.RecordSeparator, ByteStringType.Immutable, out var dbNamePrefix);
```

Comment:
are we putting items using RecordSeparator (30) or `/` (47) ?

Reply:
I used the same one as in other subscriptions methods:
https://github.com/garayx/ravendb/blob/1ed42da34be3ed8c75a32581f86db9f6951dff95/src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionsState.cs#L567

Reply:
Awesome, wanted to be sure